docs/developers/architecture/app_model.md

```diff
@@ -121,81 +286,69 @@ keybindings in a namespaced way that avoids usage of strings:
-
-
-The following code would register a new "Split RGB" command, to be added to a
-specific section of the layerlist context menu, with a `Cmd+Alt+T` keybinding.
-
-Note that while strings could be used for `id`, `title`, `menus.id` and
-`keybindings.primary`, the usage of enums and constants makes refactoring and
-maintenance much easier (and provides autocompletion in an IDE!)
-
-```python
-from app_model.types import Action, KeyMod, KeyCode
-from napari._app_model.constants import CommandId, MenuId, MenuGroup
-from napari._app_model.context import LayerListContextKeys as LLCK
-from napari._app_model import get_app
-
-
-# `layers` will be injected layer when this action is invoked
-def split_rgb_layer(layers: 'LayerList'):
-    ...
-
-
-action = Action(
-    id=CommandId.LAYER_SPLIT_RGB,
-    title=CommandId.LAYER_SPLIT_RGB.title,
-    callback=split_rgb_layer,
-    menus = [
-        {
-            'id': MenuId.LAYERLIST_CONTEXT,
-            'group': MenuGroup.LAYERLIST_CONTEXT.SPLIT_MERGE,
-            'when': LLCK.active_layer_is_rgb,
-        }
-    ],
-    keybindings=[{'primary': KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KeyT }]
-)
-
-get_app().register_action(action)
-```
-
-````{note}
-If you're following along in the console, you may see the following error
-when executing the above code:
-
-```python
-ValueError: Command 'napari:layer:split_rgb' already registered
-```
+A keybinding can be specified using the `keybindings` field in
+{class}`~app_model.types.Action`. This takes a {class}`app_model.types.KeyBindingRule`,
+which will let you provide keybindings for all three operating systems.
+
+### Keybindings in napari
+
+App-model keybindings are currently only being used for menu bar items that
+have migrated to using app-model `Action`s.
+[Pull request 6204](https://github.com/napari/napari/pull/6204) aims to
+migrate layer and viewer actions to app-model (though it is likely this pull request
+will be split into several smaller pull requests for ease of review and better git
+history). This pull request also implements [NAP-7](nap-7), which addresses
+how keybindings are dispatched according to their priority, enablement, and potential
+conflicts.
+
+Currently keybindings are not able to be edited by the user at runtime but this
+is planned. See [issue 6600](https://github.com/napari/napari/issues/6600) for more
+details.
+
+## Dependency injection and result processing
+
+Dependency injection allows to write functions using parameter type annotations,
+and then inject dependencies (arguments) into those functions at call time.
+Very often in a GUI application, you may wish to infer some command arguments from
+the current state of the application. For example, if you have menu item linked
+to a "close window", you likely want to close the *current* window.
+Dependency injection enables this by providing arguments to commands at runtime, based
+on the type annotations in the command function definition. A 'provider', a function
```

Comment:
```suggestion
on the type annotations in the command function definition. A `provider`, a function
```

Reply:
hmm not sure about this one. When should we use `'` and when should we use backtick for provider/processor ?

Reply:
I generally think of backticks for Python code, so function/method/variable/attribute names etc. (this is why I would put `app-model` and `in-and-out` in backticks, code-style)
" would then be used to indicate/emphasize a specific word being used in (or out of) a specific context vs. necessarily it's dictionary meaning. I'd avoid ' personally.

Reply:
Sorry to clarify, you would use `"provider"` instead of `'provider'` ?

Reply:
If provider is not a Python "thing", it'd use "

Reply:
Done!